lib/components/user/nav-login-button.js

```diff
@@ -0,0 +1,124 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import {
+  MenuItem,
+  NavDropdown,
+  NavItem
+} from 'react-bootstrap'
+
+/**
+ * This component displays the sign-in status in the nav bar.
+ * - When a user is not logged in: display 'Sign In' as a link or button.
+ * - When a user is logged in, display an 'avatar' (retrieved from the profile prop)
+ *   and a dropdown button so the user can access more options.
+ */
+export default class NavLoginButton extends Component {
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    links: PropTypes.arrayOf(PropTypes.shape({
+      target: PropTypes.string,
+      text: PropTypes.string.isRequired,
+      url: PropTypes.string.isRequired
+    })),
+    onSignInClick: PropTypes.func.isRequired,
+    onSignOutClick: PropTypes.func.isRequired,
+    profile: PropTypes.shape({
+      email: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      nickname: PropTypes.string,
+      picture: PropTypes.string
+    })
+  }
+
+  static defaultProps = {
+    links: null,
+    profile: null
+  }
```

Comment:
Are these default props really necessary?

Reply:
It's PropTypes documentation, a habit from OTP-UI.

Reply:
If you remove these defaultProps, nothing changes though, right? It's essentially (not exactly) the same as setting defaultProps to `undefined`.